src/app/interceptors/auth.interceptor.ts

```diff
@@ -0,0 +1,45 @@
+import type { HttpErrorResponse, HttpEvent, HttpHandler, HttpInterceptor, HttpInterceptorFn, HttpRequest } from '@angular/common/http';
+import { Injectable, inject } from '@angular/core';
+import { catchError, of } from 'rxjs';
+import { JwtHelperService } from '@auth0/angular-jwt'
+import { AuthService } from '../auth/auth.service';
+import { NotificationService } from '../services/notification/notification.service';
+import { Router } from '@angular/router';
+
+export const authInterceptor: HttpInterceptorFn = (req, next) => {
+  const notificationService = inject(NotificationService);
+
+  if(req.headers.get('skip')){
+    return next(req).pipe(
+      catchError((error) => {
+        notificationService.openSnackBar(error, "err", "Ok");
+        return of();
+      })
+    );
+  }
```

Comment:
```ts
const notificationService = inject(NotificationService);

  if(req.headers.get('skip')){
    return next(req).pipe(
      catchError((error) => {
        notificationService.openSnackBar(error, "err", "Ok");
        return of();
      })
    );
  }
```

This code belongs to different logic, and shouldn't be in `authInterceptor`. 
As I've mentioned previously, you can create as many interceptors as you want, just create another `notificationErrorInterceptor`, where you will handle all this logic related to the message displaying.

Services and most other things should only be responsible for one thing (Single Responsibility Principle).